### PR TITLE
#561 Added optional getText() argument to return limited number of document pages if set

### PR DESCRIPTION
**Pull request by alesrebec**
Solving #561 
Tested locally and working fine.

### CODE REVIEW

File: src/Smalot/PdfParser/Document.php [hunk 1]
```diff
@@ -264,11 +264,16 @@ public function getPages()
         throw new \Exception('Missing catalog.');
     }
 
-    public function getText(): string
+    public function getText($limit = NULL): string
```
Comment (k00ni) [inline src/Smalot/PdfParser/Document.php:267]:
```suggestion
    public function getText(int|null $pageLimit = null): string
```

File: src/Smalot/PdfParser/Document.php [hunk 2]
```diff
@@ -264,11 +264,16 @@ public function getPages()
...
     {
         $texts = [];
         $pages = $this->getPages();
 
+        // Only use the first X number of pages if $limit is set and numeric.
+        if (!empty($limit) && is_numeric($limit)) {
```
Comment (k00ni) [inline src/Smalot/PdfParser/Document.php:273]:
```suggestion
        if (is_int($pageLimit) && 0 < $pageLimit) {
```

**Review by k00ni — CHANGES_REQUESTED** [review]
@alesrebec thank you for this nice addition.

Please provide at least one test which covers your code.

**Comment by k00ni** [comment]
@alesrebec I took the liberty and finalized the PR. Any remarks?

I will merge it in a couple of days if there are no objections.

**Comment by alesrebec** [comment]
> @alesrebec I took the liberty and finalized the PR. Any remarks?
> 
> I will merge it in a couple of days if there are no objections.

Thanks for finalizing the PR.
I had some issues running the tests locally when I was about to write the tests and it just slipped from my todo list.
All good with the merge.

**Review by k00ni — APPROVED** [review]

**Pull request merged**